apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -0,0 +1,266 @@
+                                const value = reading[currentValue]
+                                const normalizedValue = normalizeMeterValue(value)
+
+                                if (!validateMeterValue(normalizedValue)) {
+                                    throw new GQLError(ERRORS.INVALID_METER_VALUE(i18n(`meter.import.column.${currentValue}`, { locale }), value), context)
+                                }
+
+                                return {
+                                    ...result,
+                                    [currentValue]: normalizedValue,
+                                }
+                            }, {})
+
+                            const rawControlReadingsDate = get(reading, ['meterMeta', 'controlReadingsDate'])
+
+                            const createdMeter = await Meter.create(context, {
+                                dv,
+                                sender,
+                                organization: { connect: organization },
+                                property: { connect: { id: property.id } },
+                                unitType: reading.addressMeta.unitType,
+                                unitName: reading.addressMeta.unitName,
+                                accountNumber: reading.accountNumber,
+                                number: reading.meterNumber,
+                                resource: { connect: reading.meterResource },
+                                numberOfTariffs: get(reading, ['meterMeta', 'numberOfTariffs'], Object.values(values).filter(Boolean).length),
+                                place: get(reading, ['meterMeta', 'place']),
+                                verificationDate: toISO(get(reading, ['meterMeta', 'verificationDate'])),
+                                nextVerificationDate: toISO(get(reading, ['meterMeta', 'nextVerificationDate'])),
+                                installationDate: toISO(get(reading, ['meterMeta', 'installationDate'])),
+                                commissioningDate: toISO(get(reading, ['meterMeta', 'commissioningDate'])),
+                                sealingDate: toISO(get(reading, ['meterMeta', 'sealingDate'])),
+                                controlReadingsDate: rawControlReadingsDate ? toISO(rawControlReadingsDate) : dayjs().toISOString(),
+                            })
+                            meterId = createdMeter.id
+                        } catch (e) {
+                            resultRows.push(e)
+                            continue
+                        }
+                    }
+
+                    try {
+                        const createdMeterReading = await MeterReading.create(context, {
+                            dv,
+                            sender,
+                            meter: { connect: { id: meterId } },
+                            source: { connect: { id: IMPORT_CONDO_METER_READING_SOURCE_ID } },
```

Comment:
Should there be an option here to select the source of `external_import` if it is not called from `condo`?

Reply:
We think about it when start to develop the mini-app for meters.